packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -888,8 +904,30 @@ private void destroyMapViewIfNecessary() {
     if (mapView == null) {
       return;
     }
-    mapView.onDestroy();
+
+    final MapView mapReference = mapView; // keep a reference to the mapView for the callback
     mapView = null;
+
+    // This fixes an issue where the mapView is still being used by the render thread after disposal.
+    // Delaying the actual mapView disposal to the next frame avoids the issue.
+    Runnable r =
+        () -> {
+          Log.e(TAG, "Mapview destroy in callback");
+          mapReference.onDestroy();
+        };
+
+    OnMapsSdkInitializedCallback mapCallback =
+        renderer -> {
+          if (renderer == MapsInitializer.Renderer.LATEST) {
+            Log.e(TAG, "Renderer LATEST");
+            handler.post(r);
+          } else if (renderer == MapsInitializer.Renderer.LEGACY) {
+            Log.e(TAG, "Renderer LEGACY");
+            handler.postDelayed(r, 100);
+          }
+        };
+
+    initializer.initialize(context, null, mapCallback);
```

Comment:
This is deferring destruction somewhat in all cases, since this is asynchronous. It would be cleaner to do this on map creation, storing the result, and then not checking here. (You can default to doing it if we haven't received a response, to avoid a possible race between initialization completing and deletion starting.)

Reply:
I've put the initialization in the `onCreate` function. Default behaviour is now the delayed destruction, in case the preferredRenderer variable has not been set.

Sadly I am currently having troubles with my Android Studio setup (again :sweat_smile: ) and until those are fixed, I can't really change/test the current unit tests, which are probably broken by the aforementioned change.